.circleci/config/jobs/website-docker-image.yml

```diff
@@ -0,0 +1,21 @@
+docker:
+  - image: circleci/buildpack-deps
+steps:
+  - checkout
+  - setup_remote_docker
+  - run:
+      name: Diff package-lock.json
+      command: |
+        echo 'export PACKAGE_LOCK_CHANGED=$(git diff --name-only $(git log --pretty=format:'%h' -n1 HEAD~1)...HEAD | grep -c website/package-lock.json)' >> $BASH_ENV
+  - run:
+      name: Build Docker Image if Necessary
+      command: |
+        if [ "$CIRCLE_BRANCH" = "master" ] && [ $PACKAGE_LOCK_CHANGED -gt 0 ]; then
+          cd website/
+          docker build -t hashicorp/vault-website:$CIRCLE_SHA1 .
+          docker tag hashicorp/vault-website:$CIRCLE_SHA1 hashicorp/vault-website:latest
+          docker login -u $DOCKER_USER -p $DOCKER_PASS
+          docker push hashicorp/vault-website
+        else
+          echo "Not building a new website docker image - branch is not master and/or dependencies have not changed."
```

Comment:
Just curious how long the website takes to build? From an outside system's point of view, tagging an image for every built commit could be easier to use, but if it's really slow then this seems a reasonable optimisation. I looked at the [builds for this PR](https://circleci.com/gh/hashicorp/vault/tree/pull%2F8154), but I could only find [one instance of this job](https://circleci.com/gh/hashicorp/vault/34605), and it was skipping the website build. In future, if you work on a branch on the main repo, rather than your fork, then CircleCI will start building your branch as soon as you push, leaving a useful history of builds for reviewers and giving you early warning of any issues before you open a PR.

Reply:
It doesn't take that long, but I figured it would be a bit silly and inefficient to build and release a new docker image for every single commit made to vault. This logic only builds and tags new images when the website's dependencies change.